os/kernel/log_dump/log_dump.c

```diff
@@ -505,6 +505,26 @@ int log_dump_save(char ch)
 	return LOG_DUMP_OK;
 }
 
+size_t log_dump_get_size(void)
+{
+	size_t size = 0;
+	struct log_dump_chunk_s* head = (struct log_dump_chunk_s *)sq_peek(&log_dump_chunks);
+	while (head != (struct log_dump_chunk_s *)sq_tail(&log_dump_chunks)) {
+		head = (struct log_dump_chunk_s *)sq_next((sq_entry_t *)head);
+		size += CONFIG_LOG_DUMP_CHUNK_SIZE;
+	}
```

Comment:
1. Don't you need to do lock / unlock for log dump?
2. If tail should be fixed, I think we can get it once. Not every while loop.
3. How about calculating current size at every node allocation and free in some static global variable? We already need to get current size to adjust node total size based on free heap size.